packages/ckit/src/tx-builders/pw/TransferSudtPwBuilder.ts

```diff
@@ -4,75 +4,233 @@ import { Amount, Builder, Cell, RawTransaction, Transaction } from '@lay2/pw-cor
+      optionsGroupBySudt.set(options.sudt.args, newRecipientOptions);
+    });
+
+    const senderSudtInputCells = new Array<Cell>(0);
+    const recipientSudtInputCells = new Array<Cell>(0);
+    const senderSudtOutputCells = new Array<Cell>(0);
+    const recipientSudtOutputCells = new Array<Cell>(0);
+
+    for (const item of optionsGroupBySudt) {
+      let [, optionsOfSameSudt] = item;
+      optionsOfSameSudt = this.deduplicateOptions(optionsOfSameSudt);
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      const sudtScript = optionsOfSameSudt[0]!.sudt;
+
+      let totalTransferAmount = new Amount('0', 0);
+      for (const option of optionsOfSameSudt) {
+        totalTransferAmount = totalTransferAmount.add(new Amount(option.amount, 0));
+        if (option.policy === 'findOrCreate') {
+          const recipientLiveSudtCell = (await this.provider.collectUdtCells(option.recipient, sudtScript, '0'))[0];
+          option.policy = recipientLiveSudtCell ? 'findAcp' : 'createCell';
+        }
+        switch (option.policy) {
+          case 'findAcp': {
+            const recipientLiveSudtCell = (await this.provider.collectUdtCells(option.recipient, sudtScript, '0'))[0];
+            if (!recipientLiveSudtCell) boom(`recipient ${option.recipient} has no cell to carry the udt`);
+            const recipientSudtInputCell: Cell = new Cell(
+              new Amount(recipientLiveSudtCell.output.capacity, 0),
+              Pw.toPwScript(recipientLiveSudtCell.output.lock),
+              recipientLiveSudtCell.output.type && Pw.toPwScript(recipientLiveSudtCell.output.type),
+              Pw.toPwOutPoint(recipientLiveSudtCell.out_point),
+              recipientLiveSudtCell.output_data,
+            );
+            // increase recipient sudt
+            const recipientSudtOutputCell = recipientSudtInputCell.clone();
+            recipientSudtOutputCell.setSUDTAmount(
+              recipientSudtOutputCell.getSUDTAmount().add(new Amount(option.amount, 0)),
+            );
+
+            recipientSudtInputCells.push(recipientSudtInputCell);
+            recipientSudtOutputCells.push(recipientSudtOutputCell);
+            break;
+          }
+          case 'createCell': {
+            containCreateOption = true;
+            const recipientSudtOutputCell = new Cell(
+              new Amount('1').add(new Amount(String(byteLenOfSudt()))),
+              Pw.toPwScript(this.provider.parseToScript(option.recipient)),
+              Pw.toPwScript(sudtScript),
+              undefined,
+              new Amount(option.amount, 0).toUInt128LE(),
+            );
+            recipientSudtOutputCells.push(recipientSudtOutputCell);
+            break;
+          }
+        }
+      }
+
+      const senderLiveSudtCells = await this.provider.collectUdtCells(
+        this.sender,
+        sudtScript,
+        totalTransferAmount.toHexString(),
+      );
+      const senderSudtInputCell: Cell[] = senderLiveSudtCells.map(
```

Comment:
```ts
const collectedSenderInputCells = senderLiveSudtCells.map(Pw.toPwCell);
```

- the `senderSudtInputCell` looks like a single cell, maybe we should change it
- Pw.toPwCell can convert collected cells to PwCell

Reply:
fixed